crates/fuel-core/src/database/metadata.rs

```diff
@@ -45,4 +48,19 @@ impl Database {
     pub fn get_chain_name(&self) -> DatabaseResult<Option<String>> {
         self.get(CHAIN_NAME_KEY, Column::Metadata)
     }
+
+    pub fn increase_tx_count(&self, new_txs: u64) -> DatabaseResult<u64> {
+        // TODO: how should tx count be initialized after regenesis?
+        let current_tx_count: u64 =
+            self.get(TX_COUNT, Column::Metadata)?.unwrap_or_default();
+        // Using saturating_add because this value doesn't significantly impact the correctness of execution.
+        let new_tx_count = current_tx_count.saturating_add(new_txs);
+        self.insert::<_, _, u64>(TX_COUNT, Column::Metadata, &new_tx_count)?;
+        Ok(new_tx_count)
+    }
+
+    pub fn get_tx_count(&self) -> DatabaseResult<usize> {
```

Comment:
Should this be a `u64`, since it's being inserted into the DB as a `u64`?

Reply:
Is the size bounded by `u64` or `usize` or neither? If neither, I'd say we use `u64` everywhere.

Reply:
ah yep, missed this one